src/Marten/Events/AsyncProjectionTestingExtensions.cs

```diff
@@ -13,62 +16,76 @@ public static class TestingExtensions
     /// Wait for any running async daemons to catch up to the latest event sequence at the time
     /// this method is invoked for all projections. This method is meant to aid in automated testing
     /// </summary>
-    /// <param name="store"></param>
-    /// <param name="timeout"></param>
     public static async Task WaitForNonStaleProjectionDataAsync(this IDocumentStore store, TimeSpan timeout)
     {
         var databases = await store.Storage.AllDatabases().ConfigureAwait(false);
+        var projectionsCount = store.AsyncProjectionsCount();
 
         if (databases.Count == 1)
         {
-            await WaitForNonStaleProjectionDataAsync(databases.Single(), timeout).ConfigureAwait(false);
+            await WaitForNonStaleProjectionDataAsync(databases.Single(), projectionsCount, timeout).ConfigureAwait(false);
         }
         else
         {
-            var tasks = databases.Select(db => db.WaitForNonStaleProjectionDataAsync(timeout));
+            var tasks = databases.Select(db => db.WaitForNonStaleProjectionDataAsync(projectionsCount, timeout));
             await Task.WhenAll(tasks).ConfigureAwait(false);
         }
     }
 
-
     /// <summary>
     /// Wait for any running async daemon for a specific tenant id or database name to catch up to the latest event sequence at the time
     /// this method is invoked for all projections. This method is meant to aid in automated testing
     /// </summary>
-    /// <param name="store"></param>
     /// <param name="tenantIdOrDatabaseName">Either a tenant id or the name of a database within the system</param>
-    /// <param name="timeout"></param>
     public static async Task WaitForNonStaleProjectionDataAsync(this IDocumentStore store, string tenantIdOrDatabaseName, TimeSpan timeout)
     {
         // Assuming there's only one database in this usage
         var database = await store.Storage.FindOrCreateDatabase(tenantIdOrDatabaseName).ConfigureAwait(false);
 
         if (store.Storage is DefaultTenancy)
-
-        await WaitForNonStaleProjectionDataAsync(database, timeout).ConfigureAwait(false);
+            await WaitForNonStaleProjectionDataAsync(database, store.AsyncProjectionsCount(), timeout).ConfigureAwait(false);
```

Comment:
not sure, if this code is valid.. my Rider shows 
`Suspicious type check: there is no type in the solution that is inherited from both 'Marten.IMartenStorage' and 'Marten.Storage.DefaultTenancy'`

![image](https://github.com/JasperFx/marten/assets/110024370/0e409672-5b9a-469a-ab39-2209689520a1)